### PR TITLE
Replace RocskDB with SpeeDB

### PR DESCRIPTION
**Pull request by krishnangovindraj**
## What is the goal of this PR?
Use SpeeDB in place of RocksDB as the underlying key-value store.

## What are the changes implemented in this PR?
* Update the maven artifact dependencies.
* Propagate the change to the database & server packages.
 
We have decided not to rename references of RocksDB in the code to SpeeDB, because speedb retains the rocksdb interface and the speedb-jni  API retains the rocksdb one (the package, class and function names are identical)

### CODE REVIEW

File: server/BUILD [hunk 1]
```diff
@@ -148,20 +148,17 @@ java_binary( # TODO: this target should not be needed if :server-deps-windows ca
     tags = ["maven_coordinates=com.vaticle.typedb:typedb-core-server-bin-deps:{pom_version}"],
 )
 
-java_binary( # TODO: this target should not be needed if :server-deps-prod can depend directly on @maven//:org_rocksdb_rocksdbjni (lib)
+java_binary( # TODO: this target should not be needed if :server-deps-prod can depend directly on @maven//:io_github_speedb_io_speedbjni_2_4_1 (lib)
```
Comment (krishnangovindraj) [inline server/BUILD:151]: This TODO may now be do-able.
Is there a reason we want the `server/lib/{common, prod, dev}` path structure for dependencies?

File: dependencies/vaticle/repositories.bzl [hunk 2]
```diff
@@ -20,8 +20,8 @@ load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")
 def vaticle_dependencies():
     git_repository(
         name = "vaticle_dependencies",
-        remote = "https://github.com/vaticle/dependencies",
-        commit = "51fdda8c935d5f9fc34bfdf41b0a6ac1ca77e2ee",  # sync-marker: do not remove this comment, this is used for sync-dependencies by @vaticle_dependencies
+        remote = "https://github.com/krishnangovindraj/dependencies",
+        commit = "c340321b671a013004a309d63d8d08cf83d1d1f1",  # sync-marker: do not remove this comment, this is used for sync-dependencies by @vaticle_dependencies
```
Comment (krishnangovindraj) [inline dependencies/vaticle/repositories.bzl:24]: Must update this before merge.

**Review by krishnangovindraj — COMMENTED** [review]
self-review

File: database/BUILD [hunk 3]
```diff
@@ -54,17 +54,17 @@ native_java_libraries(
     mac_deps = [
         "@maven//:com_google_ortools_ortools_darwin",
         "@maven//:com_google_ortools_ortools_darwin_java",
-        "@maven//:org_rocksdb_rocksdbjni",
```
Comment (flyingsilverfin) [inline database/BUILD:57]: how is it possible that before we could depend on an un-versioned rocksdbjni and now we need to use a versioned speedbjni?

**Review by flyingsilverfin — APPROVED** [review]

**Pull request merged**